src/v/resource_mgmt/memory_groups.cc

```diff
@@ -47,7 +47,11 @@ struct memory_shares {
 
 size_t
 compaction_memory_reservation::reserved_bytes(size_t total_memory) const {
-    return std::min(total_memory, max_bytes);
+    if (!max_limit_pct.has_value()) {
+        return std::min(total_memory, max_bytes);
+    }
+    size_t bytes_limit = total_memory * (max_limit_pct.value() / 100.0);
+    return std::min(max_bytes, bytes_limit);
```

Comment:
Ah ok I see now in the last commit how things work. Based on this calculation, do we need to make the propery be a std::optional? it seems like you'd get the same behavior if you configured the option as 100%.

Reply:
Ah good point, this was mostly copy-pasta. Will remove the std::optional

Reply:
Done